### PR TITLE
deprecate_settings: Support splitting setting

### PR DESCRIPTION
**Pull request by nalinbhardwaj**
Adds support for splitting deprecated arg into multiple new args.
Adds test for it as well.

Closes https://github.com/coala/coala/issues/5683

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [x] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [x] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `corobo mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

File: tests/bearlib/DeprecateSettingsTest.py [hunk 1]
```diff
@@ -10,8 +11,22 @@ def func(new):
...
                          'This docstring will not be lost.')
+
+    def test_multiple_args(self):
+        logger = logging.getLogger()
+
+        with self.assertLogs(logger, 'WARNING') as cm:
+            self.assertEqual(func_2(a=1), 5)
+
+            self.assertRegex(cm.output[0], 'WARNING:root:The setting `a` is '
+                             r'deprecated. Please use `[xy]` instead.')
```
Comment (jayvdb) [inline tests/bearlib/DeprecateSettingsTest.py]: > Please use `[xy]` instead

This needs to be improved.

And we also need a test to cover the other scenario:

> 'The value of `{}` and `{}` are conflicting. `{}` will be used instead.

Comment (nalinbhardwaj) [reply tests/bearlib/DeprecateSettingsTest.py]: > > Please use [xy] instead
>
> This needs to be improved.

Not sure if you/I understand correctly, but this is a regex, so the actual error is "Please use `x` instead" or "Please use `y` instead". The reason this needs a regex is that the order of args seems to depend on the version of the py(py3.4 has the message with y, py3.5 with x).

Comment (nalinbhardwaj) [reply tests/bearlib/DeprecateSettingsTest.py]: > And we also need a test to cover the other scenario:
>
> > 'The value of {} and {} are conflicting. {} will be used instead.

Added.

Comment (nalinbhardwaj) [reply tests/bearlib/DeprecateSettingsTest.py]: No, there's only a single error message. This seems to be a deliberate choice in #3204.

**Comment by jayvdb** [comment]
ack https://github.com/coala/coala/pull/5701/commits/bac641f76bd516823d2d7fcd1a5425fd91c5894f

**Comment by jayvdb** [comment]
reack https://github.com/coala/coala/pull/5701/commits/7122a125a38da442d77903fc0d45a21ce42c2af5

**Comment by jayvdb** [comment]
@gitmate-bot ff

**Pull request merged**